services/catalog/service/src/graphql/plugins/entity-localization-plugin.ts

```diff
@@ -0,0 +1,83 @@
+import { Plugin } from 'graphile-build';
+import { QueryBuilder } from 'graphile-build-pg';
+import {
+  embed,
+  gql,
+  makeExtendSchemaPlugin,
+  makePluginByCombiningPlugins,
+} from 'graphile-utils';
+
+import { Table } from 'zapatos/schema';
+import { MOSAIC_LOCALE_PG_KEY } from '../../common';
+
+const wellKnownProperties = [
+  {
+    name: 'title',
+    description: (entity: string) =>
+      `Title of the ${entity.replace('_', ' ')}.`,
+    required: true,
+  },
+  {
+    name: 'synopsis',
+    description: () => 'Short description of the main plot elements.',
+    required: false,
+  },
+  {
+    name: 'description',
+    description: () => 'Extended synopsis.',
+    required: false,
+  },
+];
+
+const EntityLocalizationExtendPlugin = (
+  entityName: string, // e.g. movie
+  localizationTable: Table, //e.g. movie_localizations
+  property: string, // e.g. ["title", "description", "synopsis"]
+): Plugin => {
+  return makeExtendSchemaPlugin((build) => {
+    const { pgSql: sql, inflection } = build;
+    const fieldDefinition = wellKnownProperties.find(
+      (x) => x.name === property,
+    );
+    const comment = fieldDefinition
+      ? `"""${fieldDefinition.description(entityName)}"""`
+      : '';
+    const required = fieldDefinition?.required ? `!` : '';
+    const capitalizedEntityName = inflection.upperCamelCase(entityName); // e.g. Movie
+    const localizations_table = sql.identifier(localizationTable);
+    const schema = sql.identifier('app_public');
+    const column = sql.identifier(property);
+    const fk = sql.identifier(`${entityName}_id`);
+    const localeKey = sql.value(MOSAIC_LOCALE_PG_KEY);
+    const condition = sql.fragment`(${schema}.${localizations_table}.locale = (SELECT pg_catalog.current_setting(${localeKey}, true)) OR ${schema}.${localizations_table}.is_default_locale IS TRUE)`;
```

Comment:
minor: why extract this condition and not include it or extract other conditions as well?